shared/chat/conversation/messages/exploding-explainer-dialog/container.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+import ExplodingExplainer from '.'
+import {type RouteProps} from '../../../../route-tree/render-route'
+import {compose, connect} from '../../../../util/container'
+import {isMobile} from '../../../../constants/platform'
+
+type OwnProps = RouteProps<{}, {}>
+
+const mapStateToProps = (state, ownProps: OwnProps) => ({})
+
+const mapDispatchToProps = (dispatch, ownProps: OwnProps) => ({
+  onBack: isMobile ? undefined : () => dispatch(ownProps.navigateUp()),
+  onCancel: () => dispatch(ownProps.navigateUp()),
```

Comment:
It looks like this is gonna be hooked up in routeTree directly given the `RouteProps<>` def above for `OwnProps`. So might as well use `putActionIfOnPath` here in case user quickly clicks more than once?

Reply:
IDK I was just copying chat/delete-history-warning. I assumed this was necessary to go back to the right place? Is there a good example of an alternative?

Reply:
I searched for it, only to realize that only FS uses this now: https://github.com/keybase/client/blob/12d06a25888741544d419c79fe2e2802eb78ca79/shared/fs/oops/container.js#L19

So perhaps there's a better thing now? Not sure; @chrisnojima is `putActionIfOnPath` considered harmful now?

Reply:
@songgao for routable components `ownProps.navigateUp` is actually `putActionIfOnPath` [here](https://github.com/keybase/client/blob/master/shared/route-tree/render-route.js#L85)

Reply:
@buoyad so does this look ok or should it just use navigateUp?

Reply:
@songgao yeah, if you want to do something other than navigate or if you're in a deep child of a routable component and don't want to deal with plumbing the closures through. All this might not be relevant soon because of the imminent nav change.